### PR TITLE
Switch to using golangci-lint instead of go-metalinter

### PR DESCRIPTION
**Pull request by paddycarver**
Following in the footsteps of our friends working on the AWS provider in terraform-providers/terraform-provider-aws#7457 and prompted by [go-metalinter's mainainer's stance](https://twitter.com/alecthomas/status/1091467762625474560), we're switching from go-metalinter to golangci-lint. This PR updates the Travis config, makefile, and lint configs to make this happen.

### CODE REVIEW

**Comment by paddycarver** [comment]
Travis failed due to lint issues that it previously didn't catch. I filed https://github.com/GoogleCloudPlatform/magic-modules/pull/1387 to resolve all those issues. When that PR is merged, these lint issues will go away.

**Comment by paddycarver** [comment]
Fixes #3014.

File: .travis.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ language: go
 go:
 - "1.11.x"
 env:
-  - GO111MODULE=off
+  - GO111MODULE=on
```
Comment (rileykarson) [inline .travis.yml]: Can you see if we're able to safely set `GOFLAGS=-mod=vendor` as well?

Comment (paddycarver) [reply .travis.yml]: I believe it's unnecessary, as the linting tool does it automatically?

Comment (rileykarson) [reply .travis.yml]: Oh not for the linting tool, just because Go will realise we're using vendored modules in that case.

**Pull request merged**